src/variant_type.rs

```diff
@@ -208,6 +213,36 @@ impl ToOwned for VariantTy {
     }
 }
 
+impl StaticType for VariantTy {
+    fn static_type() -> Type {
+        unsafe { from_glib(glib_ffi::g_variant_type_get_gtype()) }
+    }
+}
+
+impl StaticType for VariantType {
+    fn static_type() -> Type {
+        unsafe { from_glib(glib_ffi::g_variant_type_get_gtype()) }
+    }
+}
+
+impl SetValue for VariantTy {
+    unsafe fn set_value(value: &mut Value, this: &Self) {
+        gobject_ffi::g_value_take_boxed(value.to_glib_none_mut().0, this.to_glib_none().0 as glib_ffi::gpointer)
```

Comment:
Shouldn't this be `set_boxed`? The rust part takes a reference, not ownership of the variant ty